pulser/sequence.py

```diff
@@ -1081,29 +1081,49 @@ def abstract_repr(self) -> dict[str, Any]:
             ``serialize``
         """
 
-        res = {"register": [], "channels": [], "operations": []}
+        mapping = {
+            "CompositeWaveform": "composite",
+            "CustomWaveform": "custom",
+            "ConstantWaveform": "constant",
+            "RampWaveform": "ramp",
+            "BlackmanWaveform": "blackman",
+            "InterpolatedWaveform": "interpolated",
+            "KaiserWavewform": "kaiser",
+        }
+
+        res = {
+            "version": "1",
+            "register": {},
+            "channels": {},
+            "parameters": {},
+            "operations": [],
+        }
         operations = res["operations"]
         for call in self._calls:
             if call.name == "__init__":
                 register, device = call.args
                 # process register
                 res["device"] = device.name
                 for name, (x, y) in register.qubits.items():
-                    res["register"].append({"name": name, "x": x, "y": y})
+                    res["register"][name] = {"x": x, "y": y}
```

Comment:
The qubit `name` can actually be an `int`, is this a problem if being used as a key in the JSON?

Reply:
Ping in case you missed this question